File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/CreatedFulfilmentImpulse.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

/**
 * Impulse which indicates the resulf of a create fulfilment impulse
 */
class CreatedFulfilmentImpulse: Type() {

    fun id() { 
        add {"id" } 
    }

    /**
     * the fulfilment instance
     */
    fun data(config: Fulfilment.() -> Unit) = 
        add { "data{ " + Fulfilment().apply(config).render(it) + "}"}

}

File: gaia-sdk-python/gaia_sdk/graphql/response/type/CreatedFulfilmentImpulse.py

from gaia_sdk.graphql.response.type.Fulfilment import Fulfilment

Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class CreatedFulfilmentImpulse:
    """
    Impulse which indicates the result of a create fulfilment impulse
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))
    """
    the fulfilment instance
    """
    @property
    def data(self) -> Fulfilment:
        return Fulfilment(self.dictionary.get("data"))
